ChangeLog.md

```diff
@@ -1,11 +1,105 @@
+   `AZCOPY_CONCURRENT_FILES` environment variable. This is an advanced setting, which generally
+   should not be modified.  It does not affect the number of HTTP connections, which is still
+   controlled by `AZCOPY_CONCURRENCY_VALUE`.
+1. The values of key environment variables are listed at the start of the log file.
+1. An official 32-bit build is now released, in addition to the usual 64-bit builds. (TBC: For which OS's?)
+
+### Bug fixes
 
-1. The sync command's output in JSON has changed for consistency reasons. Tools integrating AzCopy should be aware.
+1. Long pathnames (over 260 characters) are now supported everywhere on Windows, including on UNC
+   shares.
+1. When supplying a `--content-type` on the command line it's no longer necessary to also specify
+   `--no-guess-mime-type`.
+1. There is now no hard-coded limit on the number of files that can be processed by the `sync`
+   command.  The number that can be processed (without paging of memory to disk) depends only on the
+   amount of RAM available.
+1. Transfer of sparse page blobs has been improved, so that for many sparse page blobs only the
+   populated pages will transferred.  The one exception is blobs which have had a very high number
+   of updates, but which still have significant sparse sections.  Those blobs may not be
+   transferred optimally in this release. Handling of such blobs will be improved in a future release.
+1. Accessing root of drive (e.g. `d:\`) no longer causes an error.
+1. On slow networks, there are no longer excessive log messages sent to the Event Log (Windows) and
+   SysLog (Linux).
+1. If AzCopy can't check whether it's up to date, it will no longer hang. (Previously, it could hang
+   if its version check URL, https://aka.ms/azcopyv10-version-metadata, was unreachable due to
+   network routing restrictions.)  
+1. (TBC: cleanup of in-flight files is improved after a cancellation (CTRL-C))
+1. High concurrency values are supported (e.g. over 1000 connections). While these values are seldom
+   needed, they are occasionally useful - e.g. for service-to-service transfer of files around 1 MB
+   in size.
+1. Files skipped due to "overwrite=false" are no longer logged as "failed".
+1. Logging is more concise at the default log level.
+1. Error message text, returned by Blob and File services, is now included in the log.
+1. A log file is created for copy jobs even when there was nothing to copy.
+1. In the log, UPLOAD SUCCESSFUL messages now include the name of the successful file.
+1. Clear error messages are given to show that AzCopy does not currently support Customer-Provided
+   Encryption Keys.
+1. On Windows, downloading a filename with characters not supported by the operating system will
+   result in those characters being URL-encoded to construct a Windows-compatible filename. The
+   encoding process is reversed if the file is uploaded. 
+1.  Uploading a single file to ADLS Gen 2 works now.
```

Comment:
Add to the end:
1. Jobs downloading from ADLS Gen 2 that result in no scheduled transfers will no longer hang.